Apps/W1/EDocumentConnectors/Tietoevry/app/src/IntegrationImpl.Codeunit.al

```diff
@@ -0,0 +1,47 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+namespace Microsoft.EServices.EDocumentConnector.Tietoevry;
+
+using System.Utilities;
+using Microsoft.EServices.EDocument;
+using Microsoft.eServices.EDocument.Integration.Send;
+using Microsoft.eServices.EDocument.Integration.Receive;
+using Microsoft.eServices.EDocument.Integration.Interfaces;
+
+codeunit 6392 "Integration Impl." implements IDocumentSender, IDocumentResponseHandler, IDocumentReceiver
+{
+    Access = Internal;
+
+    procedure Send(var EDocument: Record "E-Document"; var EDocumentService: Record "E-Document Service"; SendContext: Codeunit SendContext)
+    begin
+        this.TietoevryProcessing.SendEDocument(EDocument, EDocumentService, SendContext);
+    end;
+
+    procedure GetResponse(var EDocument: Record "E-Document"; var EDocumentService: Record "E-Document Service"; SendContext: Codeunit SendContext): Boolean
+    begin
+        exit(this.TietoevryProcessing.GetDocumentStatus(EDocument, SendContext));
+    end;
+
+    procedure ReceiveDocuments(var EDocumentService: Record "E-Document Service"; ReceivedEDocuments: Codeunit "Temp Blob List"; ReceiveContext: Codeunit ReceiveContext)
+    begin
+        this.TietoevryProcessing.ReceiveDocuments(EDocumentService, ReceivedEDocuments, ReceiveContext);
+    end;
+
+    procedure DownloadDocument(var EDocument: Record "E-Document"; var EDocumentService: Record "E-Document Service"; DocumentMetadataBlob: codeunit "Temp Blob"; ReceiveContext: Codeunit ReceiveContext)
+    begin
+        this.TietoevryProcessing.DownloadDocument(EDocument, EDocumentService, DocumentMetadataBlob, ReceiveContext);
+        this.TietoevryProcessing.AcknowledgeDocument(EDocument, EDocumentService, DocumentMetadataBlob, ReceiveContext);
```

Comment:
Implement IDocumentResponseHandler, and do that there :)

Reply:
The interface "IDocumentResponseHandler" does not include a procedure to handle acknowledge of documents collected, only a procedure the "GetResponse" of the submitted document. Is this something you can consider to implement somehow? @Groenbech96